internal/caching/cache_lazy_load_members.go

```diff
@@ -0,0 +1,71 @@
+package caching
+
+import (
+	"fmt"
+	"time"
+
+	userapi "github.com/matrix-org/dendrite/userapi/api"
+)
+
+const (
+	LazyLoadCacheName       = "lazy_load_members"
+	LazyLoadCacheMaxEntries = 128
+	LazyLoadCacheMutable    = true
+	LazyLoadCacheMaxAge     = time.Minute * 30
+)
+
+type LazyLoadCache struct {
+	// Mapping from userID/deviceID to InMemoryLRUCachePartition
+	userCaches map[string]*InMemoryLRUCachePartition
+}
+
+// NewLazyLoadCache creates a new LazyLoadCache.
+func NewLazyLoadCache() *LazyLoadCache {
+	return &LazyLoadCache{
+		userCaches: make(map[string]*InMemoryLRUCachePartition),
```

Comment:
Should this be unbounded? I was somewhat expecting that we'd have an LRU cache containing LRU partitions.